PPG-21#ValiderEtape

ibib-git opened the pull request.

ibib-git committed:
ppg-21
Constantes : ajout de "créer roles"
TacheEntity : suppression commentaires
ProjetRepository : ajout native query "getIdProjetEntityByTacheParent" & "getIdProjetEntityByTacheEnfant"
TacheRepository : ajout native query "getIdTacheParent" & "getAllTachesPrecedente" & "getAllTachePrecedenteValidee" & "getAllTacheEnfant" & "getAllTacheEnfantValidee"
CreationDeProjetService : remplacement string brut par constantes
ValiderTacheService : création du service (wip) & méthode verificationDroitUtilisateurService
UcmppgApplication : ajout notion d historique et tache validee & remplacement string brut par constantes

ibib-git committed:
ppg-21
ValiderTacheService : achevement du service principal et création d'une méthode de verificationValidationTacheSpecialHistorique() afin de check si les taches enfants/précedentes sont validées ou non
TacheDTO : Modification de la liste d id de taches précédentes en list de TacheDTO précédente et ajout de l'attribut historique dans le constructeur
TacheController : création d'un controller pour la gestion des taches et d'une méthode validerTache() ainsi que d'une méthode de gestion des cas d'erreurs depuis le service (serviceException())
UtilisateurController : modification du string de retour pour getUtilisateurParId()
ErrorServiceException : création d'une exception personnalisée pour la gestion des cas d'erreurs dans les services
EtapeWorkflowRepository : ajout de méthodes par native query
HistoriqueTacheRepository : ajout de méthodes par native query
TacheRepository : ajout de méthodes par native query
UcmppgApplication : ajout d'hostorique dans les data de base et modification de l ajout collaborateur dans projet divan et pas cheese project

ibib-git committed:
ppg-21
ValiderTacheService : correction de l idEtape utilisé poour check la validation des enfants (il faut l id de l etape suivante)
TacheController : delete println()
UcmppgApplication : modification d historique et d'ordre de save pour les taches

ibib-git committed:
Merge branch 'develop' into PPG-21#ValiderEtape

# Conflicts:
#	src/main/java/be/technobel/ucmppg/UcmppgApplication.java
#	src/main/java/be/technobel/ucmppg/dal/repositories/EtapeWorkflowRepository.java
#	src/main/java/be/technobel/ucmppg/dal/repositories/HistoriqueTacheRepository.java
#	src/main/java/be/technobel/ucmppg/dal/repositories/TacheRepository.java

ibib-git committed:
ppg-21
merge bug fixe : UcmppgApplication : duplicate code

WattecampsProg commented on src/main/java/be/technobel/ucmppg/bl/service/projet/ChangerOrdreEtapeService.java:
```diff
@@ -63,7 +63,7 @@ public Boolean execute(Long idUtilisateur, Long idEtape, int nvOrdre)
                                 .forEach(e -> e.setNumOrdreEtapeWorkflow(e.getNumOrdreEtapeWorkflow()-1));
                     }
-
+                    //TODO DAMIEN : a remplacer (Guillaume advice)
```
Explique

ibib-git replied:
Quand Guillaume a regardé mon code pour m'aider la dernière fois il m a fais remarqué que au lieu de faire deux instructions pour update le set je peux le faire en une instruction

WattecampsProg commented on src/main/java/be/technobel/ucmppg/dal/repositories/TacheRepository.java:
```diff
@@ -13,20 +14,44 @@
 @Repository
 public interface TacheRepository extends CrudRepository<TacheEntity,Long> {
 
+    @Query(value = "select ID_TACHE from TABLEAU_DE_TACHE t\n" +
```
\n sont pas utilis

ibib-git replied:
c'est le copier-coller depuis la console de db qui les a mis auto

WattecampsProg commented on src/main/java/be/technobel/ucmppg/dal/repositories/HistoriqueTacheRepository.java:
```diff
@@ -12,6 +14,24 @@
+            "                  join TACHES_PRECEDENTES tp\n" +
+            "                       on tp.TACHE_PRECEDENTE = t.ID_TACHE\n" +
+            "                where tp.TACHE_SUIVANTE = ?1)",
+            nativeQuery = true)
```
T'aime les espaces

ibib-git replied:
c'est le copier coller avec le tab

WattecampsProg commented on src/main/java/be/technobel/ucmppg/dal/repositories/HistoriqueTacheRepository.java:
```diff
@@ -12,6 +14,24 @@
+            "                on t.ID_TACHE = te.TACHES_ENFANTS_ID_TACHE\n" +
+            "            where te.TACHE_ENTITY_ID_TACHE = ?1)",
+            nativeQuery = true)
+    Optional<List<HistoriqueTacheEntity>> getAllTacheEnfantValidee (Long idTache);
```
Idem

WattecampsProg commented on src/main/java/be/technobel/ucmppg/bl/dto/projet/taches/TacheDTO.java:
```diff
@@ -23,7 +25,7 @@
     private String nom;
     private String description;
     private List<TacheDTO> tacheEnfants=new ArrayList<>();
-    private List<Long> tachesPrecedentes=new ArrayList<>();
```
j ai deja remplacé

ibib-git committed:
ppg-21
merge bug fixe : EtapeWorkflowRepository & HistoriqueTacheRepository & ProjetRepository & TacheRepository : correction query (suppress /n & add @Param clean code)

ibib-git committed:
ppg-21
merge bug fixe :ProjetDTO ajout de l id mapping dans constructeur
